### PR TITLE
Int tests interfaces

### PR DESCRIPTION
**Pull request by cachedout**
### What does this PR do?

Supercedes saltstack/salt#34275

### CODE REVIEW

**Comment by rallytime** [comment]
@cachedout This is exhibiting the same circular imports error we were seeing the other day:
```
16:29:44 Warning: Permanently added '50.116.7.88' (ECDSA) to the list of known hosts.
16:29:45 Connection to 50.116.7.88 closed.
16:29:45 Traceback (most recent call last):
16:29:45   File "/testing/tests/runtests.py", line 14, in <module>
16:29:45     from integration import TestDaemon, TMP  # pylint: disable=W0403
16:29:45   File "/testing/tests/integration/__init__.py", line 72, in <module>
16:29:45     from salt.utils.process import MultiprocessingProcess, SignalHandlingMultiprocessingProcess
16:29:45 ImportError: cannot import name MultiprocessingProcess
```

There is also a small [lint error](https://jenkins.saltstack.com/job/PR/job/salt-pr-lint-n/3297/violations/file/tests/integration/__init__.py/).

FWIW, I recently submitted a small change to `integration/__init__.py` to help clean up some of those imports. Perhaps a rebase off of current develop would be helpful.

**Comment by cachedout** [comment]
@rallytime OK. Then @twangboy has changed something since we last looked at it. It's probably in one of the more recent commits. All I did was tweak some interfaces in the config files. We'll have to investigate the other recent changes...

**Comment by rallytime** [comment]
@cachedout Ok sounds good. I thought a rebase would be useful because the import it's failing on (`MultiprocessingProcess`) isn't actually used in `integration/__init__.py` and I have removed it in #34586. But there certainly could be some other issues in the recent changes.

**Comment by cachedout** [comment]
@rallytime Oooh, good call there. Let me do one now!

**Comment by rallytime** [comment]
Go Go Jenkins!

**Comment by rallytime** [comment]
Bummer - the import error just moved to a new import. Something else is going on in some of the newer commits. There are also some lint errors still.

ping @twangboy

**Comment by cachedout** [comment]
I cannot replicate this inability to start the test suite at all locally.

**Comment by jtand** [comment]
@cachedout I was able to replicate this. I'm looking into why.

File: tests/integration/__init__.py [hunk 1]
```diff
@@ -30,6 +30,11 @@
 except ImportError:
     pass
 
+try:
+    import salt.ext.six.moves.socketserver as socketserver
+except ImportError:
+    import socketserver
```
Comment (jtand) [inline tests/integration/__init__.py]: @cachedout This needs to be moved below the salt libs. I think it has something to do with circular imports, but I don't know enough about it to say why.

Comment (rallytime) [inline tests/integration/__init__.py]: Oh, look, `import salt.ext.six.moves.socketserver as socketserver` is happening in this file already, below on line 84.

**Comment by twangboy** [comment]
https://github.com/cachedout/salt/pull/10

**Comment by cachedout** [comment]
@twangboy OK, I merged them in.

**Comment by cachedout** [comment]
Go Go Jenkins!

**Comment by rallytime** [comment]
Something isn't right in this branch. The PR from @twangboy removes the offending try/except block on line 33 and moves it to line 80 in `integration/__init__.py`, but this pull request still has the first try/except block present (now there are two of the same try/except import block.) Merge conflict perhaps?

**Comment by jtand** [comment]
Go Go Jenkins!

**Pull request merged**